id-repository/id-repository-identity-service/src/main/java/io/mosip/idrepository/identity/service/impl/IdRepoServiceImpl.java

```diff
@@ -533,11 +535,15 @@ private void updateRequestBodyData(IdRequestDTO request) throws IdRepoAppExcepti
 	@SuppressWarnings({ "rawtypes", "unchecked" })
 	protected void updateVerifiedAttributes(IdRequestDTO<T> requestDTO, DocumentContext inputData, DocumentContext dbData) throws IdRepoAppException {
 		List dbVerifiedAttributes = (List) dbData.read(DOT + VERIFIED_ATTRIBUTES);
+		boolean isV2Version=false;
 		dbVerifiedAttributes.remove(null);
-		boolean isV2Version = requestDTO.getVerifiedAttributes() instanceof Map ? true : false;
+		if (requestDTO.getVerifiedAttributes() != null) {
+			isV2Version = requestDTO.getVerifiedAttributes() instanceof Map;
+		}
 		if (dbVerifiedAttributes.isEmpty()) {
 			dbVerifiedAttributes.add(isV2Version ? new HashMap<>() : new ArrayList<>());
 		}
+		isV2Version = dbVerifiedAttributes.get(0) instanceof Map;
```

Comment:
dbVerifiedAttributes can be null, so you should handle that case. Also, you check for emptiness and then set it to an empty list. Trying to access the element at index "0" will throw an AIOB exception.

Reply:
taken care